api/app/Http/Controllers/resourceController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Resource;
+use Illuminate\Http\Request;
+
+class resourceController extends Controller
+{
+    // GET /resource
+    //returns list of all entries in table
+    public function index(){
+        return json_encode(Resource::all());
+    }
+
+    // GET /resource/create
+    public function create(){
+
+    }
+
+    // POST /resource
+    // updates/creates resource
+    public function store(Request $request){
+        Resource::updateOrCreate(
+            ['name' => $request->input('name'),
+                'user_id' => $request->input('user_id')],
+            ['abstract' => $request->input('abstract'),
+                'votes' => $request->input('votes')] // New resource therefore no one has voted for it yet
+        );
+        return "successfully added resource";
```

Comment:
Sorry for the spam.  I didn't mean to use my regular github account for school stuff.

Could you return the resource as json.  For new objects it'll create an id which is useful.

We don't need string messages since a 2xx status tells us it was successful.  We can use 4xx for a bad request.  If you don't want to return the object as json for any reason, use the 204 response code and leave the body empty.

Reply:
do you want the whole object or just the id returned?